src/dotnet/Fable.Tools/ProjectCracker.fs

```diff
@@ -190,18 +190,28 @@ let crackFsproj (projFile: string) =
         |> List.filter (fun x -> not(x.EndsWith("Fable.Core.dll")))
         |> List.map (fun x -> Path.Combine(projDir, Path.normalizePath x) |> Path.GetFullPath) }
 
-let rec findPaketDependenciesDir dir =
-    if File.Exists(Path.Combine(dir, "paket.dependencies")) then
+let rec findPaketDependenciesDir dir searchedDirs = 
+    printfn "findPaketDependenciesDir %s %A" dir searchedDirs
+    let path = Path.Combine(dir, "paket.dependencies")
+    if File.Exists(path) then
+        printfn "Found %s inside %s" path dir
         dir
     else
         match Directory.GetParent(dir) with
-        | null -> failwith "Couldn't find paket.dependencies directory"
-        | parent -> findPaketDependenciesDir parent.FullName
+        | null -> 
+            let errorMessage = 
+                let dirs = String.concat "\n" searchedDirs
+                sprintf "Couldn't find paket.dependencies directory, searched in: \n%s" dirs
+            failwith errorMessage
+        | parent -> 
+            let searched = searchedDirs |> List.append [dir]
+            findPaketDependenciesDir parent.FullName searched
```

Comment:
Yeah the cons `::` operator is much faster (`O(1)`). Because this function will only run a list of 5 - 10 elements, I didn't put much thought in it :smile: fixed!